betty/cropper/models.py

```diff
@@ -67,21 +63,28 @@ def optimize_image(image):
             except (TypeError, ValueError) as e:
                 # Maybe the image already had an invalid quant table?
                 if e.message.startswith("Not a valid numbers of quantization tables"):
+                    out_buffer = io.BytesIO()  # Make sure it's empty after failed save attempt
```

Comment:
Playing it safe in case out_buffer was partially written to during the (failed) im.save() call above. It's much easier to just create a new io.BytesIO() instance than to clear/reset it.